dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceFormatter.java

```diff
@@ -1001,6 +996,53 @@ public String formatObj(Object objs) {
         return ans;
     }
 
+    private String formatObject(Object objs, int depth) {
+        if (objs == null) {
+            return null;
+        }
+        if (depth >= COLLECTION_DEPTH_LIMIT) {
+            return "Collection depth limit exceeded";
+        }
+        if (objs instanceof Collection) {
+            try {
```

Comment:
Should this be changed to do something more like this?
```
if (objs instanceof Collection) {
    Object[] objArray = null;
    while (objArray == null) {
        try {
            objArray = objs.toArray();
        } catch (ConcurrentModificationException cme) {
            // this exception is possible.  Need to retry until it doesn't happen any longer.
        } catch (NoSuchElementException nsee) {
            // this exception is possible.  Need to retry until it doesn't happen any longer.
        } catch (Throwable t) {
            return "[Caught " + t.toString() + " while logging collection type " + objs.getClass().getName() + "]";
        }
    }
    StringBuilder sb = new StringBuilder("[");
    if (objArray.length != 0) {
        sb.append(formatObject(objArray[0], depth + 1));
    }
    for (int i = 0; i < objArrays.length; ++i) {
        sb.append(', ').append(formatObject(objArray[i], depth + 1));
    }
    sb.append("]");
    return sb.toString();
}
```

Reply:
done - also added an "out" if >= 100 retryable exceptions occur on the same collection